app/code/Magento/Swatches/view/adminhtml/web/js/visual.js

```diff
@@ -138,9 +138,7 @@ define([
                             elementFlags[0].value = 1;
                         }
 
-                        element.addClassName('no-display');
-                        element.addClassName('template');
-                        element.hide();
+                        element.remove();
```

Comment:
Hi @enriquei4 
After this change you can not delete existing option from attribute in admin.

Reply:
Ok, Im working to fix.

Reply:
I recommend you to change validate behavior in:
https://github.com/magento/magento2ce/blob/2.2-develop/app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php#L112
skip deleted option values:
```
        $valueOptions = (isset($options['value']) && is_array($options['value'])) ? $options['value'] : [];
        // skip deleted options
        foreach ($valueOptions as $key => $val) {
            if (isset($options['delete'][$key]) && $options['delete'][$key]) {
                unset($valueOptions[$key]);
            }
        }
```